vision_stack/task_bin/src/bin_server.cpp

```diff
@@ -2,45 +2,47 @@
 #include <task_bin/bin_server.h>
 #include <resources/topicHeader.h>
 
-Bin::Bin(std::string name, int t=0): _it(_n), _s(_n, name, boost::bind(&Bin::executeCB, this, _1), false), _actionName(name)
+char* const image_window = "Source Image";
+char* const result_window = "Result window";
+
+Bin::Bin(std::string name, int t): _it(_n), _s(_n, name, boost::bind(&Bin::executeCB, this, _1), false), _actionName(name)
 {
-	_sub = _it.subscribe(topics::CAMERA_FRONT_RAW_IMAGE, 1, &Bin::imageCallBack, this);
-  _pub = _it.advertise("/kraken/temp/", 1); // TODO: Create a topic in resources.
-  switch(t)
-  {
-    case 0:
-    {
-      templ = imread("RoboSub2015_silhouettes1.png");
-      break;
-    }
-    case 1:
-    {
-      templ = imread("RoboSub2015_silhouettes2.png");
-      break;
-    }
-    case 2:
-    {
-      templ = imread("RoboSub2015_silhouettes3.png");
-      break;
-    }
-    case 3:
-    {
-      templ = imread("RoboSub2015_silhouettes4.png");
-      break;
-    }
-  }
-  _s.start();
+	_sub = _it.subscribe(topics::CAMERA_BOTTOM_RAW_IMAGE, 1, &Bin::imageCallBack, this);
+  	_pub = _it.advertise(topics::CAMERA_BOTTOM_BIN_IMAGE, 1);
+  	switch(t)
+  	{
+    	case 0:
+    	{
+      		templ = imread("/home/kalyan/rosbuild_ws/kraken_3.0/vision_stack/task_bin/RoboSub2015_silhouettes1.png");
```

Comment:
Initially i was using relative path refering to task_buoy. It did not work.
Tried many other ways too.
ᐧ

On Mon, May 18, 2015 at 11:51 PM, Siddharth Kannan <notifications@github.com
> wrote:

> In vision_stack/task_bin/src/bin_server.cpp
> <https://github.com/auviitkgp/kraken_3.0/pull/59#discussion_r30533723>:
>
> > -      break;
> > -    }
> > -    case 3:
> > -    {
> > -      templ = imread("RoboSub2015_silhouettes4.png");
> > -      break;
> > -    }
> > -  }
> > -  _s.start();
> > +	_sub = _it.subscribe(topics::CAMERA_BOTTOM_RAW_IMAGE, 1, &Bin::imageCallBack, this);
> > +  	_pub = _it.advertise(topics::CAMERA_BOTTOM_BIN_IMAGE, 1);
> > +  	switch(t)
> > +  	{
> > +    	case 0:
> > +    	{
> > +      		templ = imread("/home/kalyan/rosbuild_ws/kraken_3.0/vision_stack/task_bin/RoboSub2015_silhouettes1.png");
>
> @kalyan-kumar <https://github.com/kalyan-kumar>
> This should have the relative path of the image, rather than the absolute
> path.
>
> Refer to how the task_buoy package references the threshold.th file. Even
> there a similar problem was being faced.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/auviitkgp/kraken_3.0/pull/59/files#r30533723>.
>